pontoon/contributors/static/js/profile.js

```diff
@@ -127,9 +136,171 @@ var Pontoon = (function (my) {
+            weekHTML = `<g transform="translate(${gx.toString()}, 0)">`;
+          }
+
+          var monthInDay = startDate.getMonth();
+
+          if (startDate.getDay() == 0 && monthInDay != usingMonth) {
+            usingMonth = monthInDay;
+            monthPosition.push({ monthIndex: usingMonth, x: gx });
+          }
+
+          var count = contributions[startDate.getTime()] || 0;
+
+          // Pick color based on count range
+          var colors = ['#333941', '#41554c', '#4f7256', '#64906D', '#7bc876'];
+          var color = colors[4];
+          switch (true) {
+            case count === 0:
+              color = colors[0];
+              break;
+            case count < 10:
+              color = colors[1];
+              break;
+            case count < 25:
+              color = colors[2];
+              break;
+            case count < 50:
+              color = colors[3];
+          }
+
+          var y = startDate.getDay() * step;
+          var date = startDate.getTime();
+          weekHTML += `<rect class="day" width="10" height="10" y="${y}" fill="${color}" data-count="${count}" data-date="${date}" rx="2" ry="2"/>`;
+
+          if (startDate.getDay() == 6) {
+            weekHTML += '</g>';
+            graphHTML += weekHTML;
+            weekHTML = null;
+            week++;
+            gx = week * step;
+          }
+
+          startDate.setUTCDate(startDate.getUTCDate() + 1);
+        }
+
+        // Add week items
+        if (weekHTML != null) {
+          weekHTML += '</g>';
+          graphHTML += weekHTML;
+        }
+
+        // Remove the first incomplete month label
+        if (monthPosition[1].x - monthPosition[0].x < 40) {
+          monthPosition.shift(0);
+        }
+
+        // Add month labels
+        for (var x = 0; x < monthPosition.length; x++) {
+          var item = monthPosition[x];
+          var monthName = monthNameFormat.format(
+            new Date(2022, item.monthIndex),
+          );
+          graphHTML += `<text x="${item.x}" y="-5" class="month">${monthName}</text>`;
+        }
+
+        // Add day labels
+        graphHTML += `
+            <text text-anchor="middle" class="wday" dx="695" dy="22">M</text>
+            <text text-anchor="middle" class="wday" dx="695" dy="48">W</text>
+            <text text-anchor="middle" class="wday" dx="695" dy="74">F</text>`;
+
+        graph.html(`
+            <svg width="690" height="110" viewBox="0 0 702 110" class="js-calendar-graph-svg">
+              <g transform="translate(0, 20)">${graphHTML}</g>
+            </svg>`);
+
+        // Handle tooltip
+        $(graph)
+          .find('.day')
+          .hover(
+            function (e) {
+              var count = $(e.target).attr('data-count');
+              var date = shortDateFormat.format($(e.target).attr('data-date'));
+              var action = count == 1 ? 'contribution' : 'contributions';
+              var text = `${count} ${action} on ${date}`;
```

Comment:
Alas, if only we had a localization system that could handle this sort of stuff. 😇

Reply:
:-)

We don't use it outside the translate app, yet.